### DISCUSSION TITLE
Adding Docker Hub OCI repository for multiple charts

### DISCUSSION THREAD

**Discussion by heruan** [artifacthub/hub#3632] [Q&A] [answered]
Hello! I'm publishing my charts to Docker Hub as described here: https://helm.sh/docs/topics/registries/ e.g.
```
helm push foo-0.1.0.tgz oci://registry-1.docker.io/heruan
helm push bar-0.1.0.tgz oci://registry-1.docker.io/heruan
```
and I see the charts published in Docker Hub.

Now I'd like to add this chart registry as a repository in ArtifactHUB, so I create a new repository targeting `oci://registry-1.docker.io/heruan` but I soon get this error message:
```
error getting repository remote digest: GET https://registry-1.docker.io/v2/heruan/tags/list?n=1000: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:heruan Type:repository]]
```

So I now wonder, is it possible to use OCI registries such as Docker Hub to act as repositories for Helm charts in ArtifactHUB?

**Comment by tegioz** [comment, marked as answer]
Hi @heruan 👋

Yes, it's possible, but you need to add one repository per chart. You can find more information about this in the [Helm charts repositories guide](https://artifacthub.io/docs/topics/repositories/helm-charts/#oci-support).

**Reply by heruan** [reply]
Thank you @tegioz for the quick answer! Oh, I stumbled upon that guide but I didn't get I'd need one repository per chart. Is that a technical limitation? It kind of defeats the purpose of having a repository if I need one for each chart 🤔

**Reply by tegioz** [reply]
No worries!

The reason it works this way is a combination of how Helm OCI based repositories work (please see [these](https://github.com/helm/helm/issues/12322) [issues](https://github.com/helm/helm/issues/11000) for more context) and that the OCI distribution spec does not (well, *it didn't* when this feature was added 😅) define a mechanism to list all references for a given namespace. We [discussed this in more detail some time ago](https://github.com/artifacthub/hub/issues/1685#issuecomment-971596259).

Artifact Hub needs an entry point to start indexing content, like it does on HTTP based Helm charts repositories using the `index.yaml` file. But AFAIK we cannot neither get this index from a special OCI reference nor discover them via OCI registry APIs. So unfortunately we need to ask publishers to provide this information to us. Hopefully this will improve over time 🙂

For OLM operators, for example, there is a special OCI reference (the catalog), that acts as an index of all the content available. This allows us to index multiple [OLM operators in an OCI registry using a single repository](https://artifacthub.io/docs/topics/repositories/olm-operators/#olm-oci-experimental-support).
